### PR TITLE
Fixing Hubble ServiceMonitor k8s-app label

### PR DESCRIPTION
**Pull request by guilhermef**
The current ServiceMonitor is trying to match a `k8s-app=cilium`, but the current service has `k8s-app=hubble`.
https://github.com/cilium/cilium/blob/master/install/kubernetes/cilium/templates/cilium-agent-service.yaml#L56

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

Fixes: #11886

### CODE REVIEW

**Review by fristonio — APPROVED** [review]
LGTM, thanks for pointing this out :100:

**Review by gandro — APPROVED** [review]
Thanks! Marking for backport to 1.9, as this seems to have regressed in the Great Helm refactor for Cilium 1.9.

**Review by kaworu — APPROVED** [review]

**Comment by kAworu** [comment]
> Fixes: #11886

@guilhermef this is referencing a closed PR where I would have expected an open issue instead. Whas that intended?

**Comment by guilhermef** [comment]
> > Fixes: #11886
> 
> 
> 
> @guilhermef this is referencing a closed PR where I would have expected an open issue instead. Whas that intended?

Yes, that's a regression, it was fixed on the referenced PR before

**Review by christarazi — APPROVED** [review]

File: install/kubernetes/cilium/templates/cilium-agent-servicemonitor.yaml
Comment (christarazi) [inline install/kubernetes/cilium/templates/cilium-agent-servicemonitor.yaml:39]: It looks like the commit that regressed is actually e9cb43c031, not the PR referenced in the commit. Could you amend the commit to:

```
Fixes: e9cb43c031 ("Helm: full refactor of helm charts, default values implemented, tests updated, kind cni integration")
```

While we're at it, could you amend the commit title to use the [imperative mood](https://chris.beams.io/posts/git-commit/#imperative), so in this case it would be `Fix`.

**Review by christarazi — CHANGES_REQUESTED** [review]
Change looks good to me; a comment on the commit itself.

File: install/kubernetes/cilium/templates/cilium-agent-servicemonitor.yaml
Comment (guilhermef) [reply install/kubernetes/cilium/templates/cilium-agent-servicemonitor.yaml:39]: @christarazi I've updated the commit message with the changes.

**Comment by christarazi** [comment]
test-me-please

**Review by christarazi — APPROVED** [review]

**Comment by PhilipSchmid** [comment]
@guilhermef Thanks for your PR :tada:. I just run into the same issue and also found the wrong label matching - but you now took the work from me to create a PR - thanks!

Would be awesome to see this PR merged in the next release 1.9.2.

Regards,
Philip

**Comment by christarazi** [comment]
net-next hit https://github.com/cilium/cilium/issues/13071. All other CI has passed and we have code owner approval. Merging.

**Pull request merged**
